tcp.c

```diff
@@ -561,6 +522,28 @@ tcp_get_address()
 	return ipaddr;
 }
 
+int
+tcp_resolve_address(char *address, uint16 port, struct addrinfo **result)
+{
+	int res;
+	char service[16];
+	struct addrinfo hints={0};
+
+	snprintf(service, sizeof(service), "%d", port);
+
+	hints.ai_flags = AI_NUMERICSERV | AI_CANONNAME;
+	hints.ai_socktype = SOCK_STREAM;
+
+	res = getaddrinfo(address, service, &hints, result);
```

Comment:
Where do we free this? Or do we let it leak on each resolve?

Reply:
I left it out because there will theoretically only be two allocations for a instance lifetime, one for commandline servername and one for redirection, both done in main()

Reply:
Only a problem if you want to run valgrind then. :)